omnibot/processor.py

```diff
@@ -311,23 +311,27 @@ def _handle_post_message(message, kwargs):
     except json.decoder.JSONDecodeError:
         logger.exception('JSON decode failure when parsing {}'.format(kwargs))
         return
-    logger.debug(ret, extra=message.event_trace)
-    if not ret['ok']:
+    except:
         logger.error(ret, extra=message.event_trace)
+        return
+    logger.debug(ret, extra=message.event_trace)
 
 
 def _handle_action(action, container, kwargs):
     parse_kwargs(kwargs, container.bot, container.event_trace)
-    ret = slack.client(
-        container.bot,
-        client_type='oauth_bot'
-    ).api_call(
-        action,
-        json=kwargs
-    )
-    logger.debug(ret)
-    if not ret['ok']:
-        if ret.get('error') == 'missing_scope':
+    try:
+        ret = slack.client(
+            container.bot,
+            client_type='oauth_bot'
+        ).api_call(
+            action,
+            json=kwargs
+        )
+        logger.debug(ret)
+    except Exception as e:
+        logger.exception('Error in handle action', kv={'error': str(e)})
+        # TODO (shekharkhedekar) validate errors
+        if str(e) == 'missing_scope':
```

Comment:
I think you need `e.response.get('error')`

Reply:
updated